### PR TITLE
remove call to system-wide gunzip. Use python version instead. 

### LINKED ISSUE #1538: gzip was not found on your system!
I use `astroquery` in my infrared spectroscopy package [radis](https://github.com/radis/radis) to retrieve HITRAN data.  

However, on each first import I get a printed message (originates from [astroquery.utils.system_tools.py](https://github.com/astropy/astroquery/blob/master/astroquery/utils/system_tools.py) )  

> gzip was not found on your system! You should solve this issue for astroquery.eso to be at its best!
On POSIX system: make sure gzip is installed and in your path!On Windows: same for 7-zip (http://www.7-zip.org)!

I do not use the `astroquery.eso` module and thus I'd like not to have to install gzip, nor to see that message. 
There are two easy options:

1. The gzip check could be moved within the `astroquery.eso` function calls.  
2. The `print` can be replaced with a Warning, which I can catch and filter from my own code. 

Option 1. requires some knowledge of `astroquery` which I don't have. Option 2. does not and I can do a MR.  

What do you think?

### PR DESCRIPTION
**Pull request by erwanp**
Fixes #1538

### CODE REVIEW

**Comment by erwanp** [comment]
@keflavich  you may want to have a look!

**Comment by keflavich** [comment]
Looks good at a glance.  Do we have any tests for this?  If not, are there any tests you could add?

Thanks!

**Comment by erwanp** [comment]
The gunzip utility is only used by the `eso` module. Could you tag anyone from there that could provide test files?

**Comment by keflavich** [comment]
I'm not sure we have any regular contributors to the ESO package at the moment, but that's a good thought.  Maybe we can at least make some sort of round-trip test to create, zip, and unzip a file... otherwise we need to come up with some example from ESO that returns a zip.  I don't know of any off the top of my head.

**Comment by erwanp** [comment]
- I ran the ESO tests locally but I didn't get any zipped file there. 
- I added a test_system_tools.py file that does a round-trip test with a dummy file
- I don't know if there is a dedicated way in `pytest` to remove dummy files, so I did it with the usual `os/os.path` functions

File: astroquery/utils/tests/test_system_tools.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+from os.path import exists
+
+# THIRD-PARTY
+import pytest
+
+# LOCAL
+from ..system_tools import gunzip
+
+def test_gunzip():
+    filename = 'test_system_tools.txt.gz'
```
Comment (keflavich) [inline astroquery/utils/tests/test_system_tools.py]: you can use `tempfile` instead, e.g.:
```
filehandle = tempfile.NamedTemporaryFile(suffix='.txt.gz')
filename = filehandle.name
```

Comment (erwanp) [reply astroquery/utils/tests/test_system_tools.py]: ty, done!

**Comment by keflavich** [comment]
We just have some minor style fixes (see https://github.com/astropy/astroquery/pull/1613#issuecomment-572549540 above, which was edited for your last PR) to get travis to go green, then I'll merge

**Comment by keflavich** [comment]
OK this is weird - @bsipocz , could you have a look?  `pep8speaks` reports no errors, but travis is reporting:
[`astroquery/utils/system_tools.py:14:1: E302 expected 2 blank lines, found 1`
](https://travis-ci.org/astropy/astroquery/jobs/636380994#L2996)

Is this a bug in our configurations, i.e., is pep8 ignoring something that travis isn't?

**Comment by keflavich** [comment]
There's a travis failure that appears unrelated to this PR.  I think this is good to merge, but let's hear from @bsipocz first to see if she has any ideas about the travis install failure

File: astroquery/utils/system_tools.py [hunk 2]
```diff
@@ -50,9 +25,16 @@ def gunzip(filename):
         Name of the decompressed file (or input filename if gzip is not
         available).
     """
-    # ".fz" denotes RICE rather than gzip compression
-    if __is_gzip_found and not filename.endswith('.fz'):
-        subprocess.call(_unzip_cmd(filename), stdout=DEVNULL, stderr=DEVNULL)
+    import shutil
+    import gzip
+
+    # @EP 09/01/20: system-wide 'gzip' was removed, Python gzip used instead.
```
Comment (bsipocz) [inline astroquery/utils/system_tools.py]: no need to date/sign comments. The git history/blame is more reliable to trace things back.

File: astroquery/utils/tests/test_system_tools.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""Test `astroquery.utils.system_tools`.
+
+"""
+
+# STDLIB
+import gzip
```
Comment (bsipocz) [inline astroquery/utils/tests/test_system_tools.py]: A bit bike shedding, but now that gzip is assumed to be optional, we need to make sure to skip the test if it's not available.

File: astroquery/utils/tests/test_system_tools.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+import os
+from os.path import exists
+import tempfile
+
+# THIRD-PARTY
+import pytest
```
Comment (bsipocz) [inline astroquery/utils/tests/test_system_tools.py]: unused import.

**Comment by bsipocz** [comment]
I'm now a bit confused. Is the python `gzip` is available on all platforms? This seems to assume it's not.

**Comment by bsipocz** [comment]
The travis failure was unrelated. The one remaining failure will be addressed by #1607 

Also, my commit might not be neccessary, but if we revert it, then the imports also need to be moved up to the top of the file, and some more comments removed, too.

**Comment by bsipocz** [comment]
The windows test failure is related, maybe @pllim has some insights.

**Comment by pllim** [comment]
Re: Windows -- I think it is the same issue that @astrofrog encountered (and fixed) in astropy/astropy#9874

**Comment by erwanp** [comment]
> 
> 
> Re: Windows -- I think it is the same issue that @astrofrog encountered (and fixed) in [astropy/astropy#9874](https://github.com/astropy/astropy/pull/9874)

The error is a Permission error on the Windows version of Travis so it doesn't seem related to  [astropy/astropy#9874](https://github.com/astropy/astropy/pull/9874) (which was pytest related). Details:

> PermissionError: [Errno 13] Permission denied: 'C:\\Users\\travis\\AppData\\Local\\Temp\\tmp0yd2zq3f.txt.gz'

I looked up for Permissions + Windows Travis-cI for for about 40min with no luck. However, I found another ocurrence of tempfile in `test_utils/test_filecontainer_save()` where a temp folder is used instead of a tempfile, and that seems to be executed with no problem.

I re-wrote our test here to use a tempfolder, we'll see if that helps.

**Comment by erwanp** [comment]
Ended up with a ```KeyError: 'independent_celestial_slices'``` error that does not seem related to this Issue. I tried to merge the latest changes  (will clean the commits at the end) but that didn't help neither. On hold until further notice.

**Comment by astrofrog** [comment]
The ``KeyError: 'independent_celestial_slices'`` error comes from reproject and should be fixed in reproject 0.7

**Comment by bsipocz** [comment]
@erwanp - could you please rebase? That would get rid of the merge commit, and retrigger CI. (I'm still not certain why the old version of reproject is picked up, rather than the new 0.7)

**Comment by erwanp** [comment]
Done, still get the reproject error.

**Comment by bsipocz** [comment]
Thank you @erwanp!

**Comment by saimn** [comment]
While looking at reading `.fits.Z` with astropy (https://github.com/astropy/astropy/issues/10714) I remembered about this and was curious to see how astroquery managed those files.
- Before it was done by `gzip` which knows how to decompress `.Z` files.
- But now, Python's `gzip` cannot decompress `.Z` files so this is broken. Testing with a file from #1580 raises a `BadGzipFile: Not a gzipped file (b'\x1f\x9d')` exception.

**Comment by bsipocz** [comment]
thanks for checking @saimn! Would you mind opening an issue about it?

**Pull request merged**